ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -1560,6 +1554,32 @@ public static ColStatistics getColStatisticsFromExpression(HiveConf conf, Statis
         }
       }
 
+      if (conf.getBoolVar(ConfVars.HIVE_STATS_USE_UDF_ESTIMATORS)) {
+        Optional<IStatEstimatorProvider> sep = engfd.getGenericUDF().adapt(IStatEstimatorProvider.class);
+        if (sep.isPresent()) {
+          Optional<IStatEstimator> se = sep.get().getStatEstimator();
+          if (se.isPresent()) {
+            List<ColStatistics> csList = new ArrayList<ColStatistics>();
+            for (ExprNodeDesc child : engfd.getChildren()) {
+              ColStatistics cs = getColStatisticsFromExpression(conf, parentStats, child);
+              if (cs == null) {
+                break;
+              }
+              csList.add(cs);
+            }
+            if(csList.size() == engfd.getChildren().size()) {
+              Optional<ColStatistics> res = se.get().estimate(csList);
+              if (res.isPresent()) {
```

Comment:
When could this happen? Maybe when column stats has anything unexpected? Could we add a comment clarifying it?

Reply:
Makes sense... Let's leave a comment in the code (maybe in the interface since it is behavior that every implementation should be aware of? or here?).